lib/phoenix_client/socket.ex

```diff
@@ -301,4 +302,18 @@ defmodule PhoenixClient.Socket do
     state = %{state | ref: ref, queue: :queue.in(push, state.queue)}
     {push, state}
   end
+
+  defp headers(opts) when is_list(opts) do
+    if dynamic_headers?(opts) do
```

Comment:
Instead of fetching both times the headers, this could be something like:

```elixir
opts |> Keyword.get(:headers) |> transform_headers()

# ...

defp transform_headers(f) when is_function(f), do: f.()
defp transform_headers(headers), do: headers
```

Reply:
I like this idea. Curious @mobileoverlord 's opinion